fem/bilininteg.cpp

```diff
@@ -615,19 +617,20 @@ void DiffusionIntegrator::AssembleElementMatrix2(
 #ifdef MFEM_THREAD_SAFE
    DenseMatrix dshape(tr_nd, dim), dshapedxt(tr_nd, spaceDim);
    DenseMatrix te_dshape(te_nd, dim), te_dshapedxt(te_nd, spaceDim);
-   DenseMatrix invdfdx(dim, spaceDim);
+   DenseMatrix invdfdx(spaceDim, spaceDim);
    Vector D(VQ ? VQ->GetVDim() : 0);
 #else
    dshape.SetSize(tr_nd, dim);
    dshapedxt.SetSize(tr_nd, spaceDim);
    te_dshape.SetSize(te_nd, dim);
    te_dshapedxt.SetSize(te_nd, spaceDim);
-   invdfdx.SetSize(dim, spaceDim);
+   invdfdx.SetSize(spaceDim, spaceDim);
    D.SetSize(VQ ? VQ->GetVDim() : 0);
 #endif
    elmat.SetSize(te_nd, tr_nd);
 
    const IntegrationRule *ir = IntRule ? IntRule : &GetRule(trial_fe, test_fe);
+   DenseMatrix te_dshapedxt_m(te_nd, spaceDim);
```

Comment:
Same as the above, i.e., move inside the `if (MQ)` statement

Reply:
Done